tracer.go

```diff
@@ -209,20 +209,31 @@ const (
 
 // SpanReference is a StartSpanOption that pairs a SpanReferenceType and a
 // referee SpanContext ("referee" is "the one who is referred to"). See the
-// SpanReferenceType documentation.
+// SpanReferenceType documentation for supported relationships.
+// If SpanReference is created with Referee == nil, it has no effect. Thus
+// it allows for a more concise syntax for starting spans:
+//
+//     sc, _ := tracer.Extract(someFormat, someCarrier)
+//     span := tracer.StartSpan("operation", opentracing.ChildOf(sc))
+//
+// The `ChildOf(sc)` option above will not panic if sc == nil, it will just
+// not add the parent span reference to the options.
 type SpanReference struct {
 	Type    SpanReferenceType
 	Referee SpanContext
 }
 
 // Apply satisfies the StartSpanOption interface.
 func (r SpanReference) Apply(o *StartSpanOptions) {
-	o.References = append(o.References, r)
+	if r.Referee != nil {
+		o.References = append(o.References, r)
+	}
 }
 
 // ChildOf returns a StartSpanOption pointing to a dependent parent span.
+// If sm == nil, the option has no effect.
 //
-// See ChildOfRef
+// See ChildOfRef, SpanReference
 func ChildOf(sm SpanContext) SpanReference {
```

Comment:
def should be `sc` -- that must have survived from the `SpanMetadata` days of the change.